qiskit/circuit/quantumcircuit.py

```diff
@@ -2565,8 +2565,19 @@ def add_calibration(self, gate, qubits, schedule, params=None):
         """
         if isinstance(gate, Gate):
             self._calibrations[gate.name][(tuple(qubits), tuple(gate.params))] = schedule
+            _name = gate.name
         else:
             self._calibrations[gate][(tuple(qubits), tuple(params or []))] = schedule
+            _name = gate
+
+        # add parameters to the _parameter_table
```

Comment:
There are some problems here. I think you will easily resolve it by writing some new test cases. Start with "steps to reproduce the problem" in issue #5524

Reply:
Hi no problem. I'm here to learn and any advice or indication is more than welcome. I'm sorry that my PR resulted in not being il line with the best practices.
Actually, it was not my intention to solve both issues with this PR. I was going after the #5524 only. 
As I start, I had only changed the `add_calibration` function in  `quantumcircuit.py` in order to update the `_parameter_table` as you had indicated in the issue. The problem is that, running the test cases, I soon found out that just adding the calibration parameter to the table was not enough to get the automated tests going. This update was causing other sorts of errors that I was able to fix by rewriting the `_expand_parameters` in `assemble.py`.  
If you take a look at #5467, you'll see that this problem impacts `_expand_parameters` as well.
So, at least from my understanding, #5467 is fired by the changes in `_expand_parameters` while #5524 prefers also the fix to #5467.
Anyways, I acknowledge that the changes are consistent and therefore need a substantial review and guidance from someone that has more experience than me.
What I can do is change the committed code as per your suggestions and work on the additional test cases. I'm not sure on how to split this into two separate PRs and I would gladly take some advice not this. I don't think that just removing the reference to the #5467 as a fixed issue helps.
Let me know, thanks a lot.

Reply:
I think I have sorted it out. Focusing only on #5524 now. I'm converting the request to draft status in order to insert the test case as well.